gems/aws-sdk-core/lib/aws-sdk-core/xml/error_handler.rb

```diff
@@ -6,7 +6,7 @@ class ErrorHandler < Seahorse::Client::Handler
 
       def call(context)
         @handler.call(context).on(300..599) do |response|
-          response.error = error(context)
+          response.error = error(context) unless response.error
```

Comment:
Because this is on 300...599 would this case ever happen? I thought because errors are for 200..299?

Reply:
This handler happens after the Http200Errors handler, which in the case of an empty body sets the status to 500.  The <Error> case in the body was actually working because this plugin creates the error from the code, but in the case of no body, it was overwriting the error with a generic error (which causes the spec to fail).